### DISCUSSION TITLE
KMK Config for that board for RP2040

### DISCUSSION THREAD

**Discussion by wajdzik** [sevmeyer/chrumm-keyboard#1] [Show and tell]
Not yet tested on an actual board, just with couple of switches and did few changes to original design. 

Also missing settings rotary bit - will work on that after I'll finish with soldering everything

```python
print("Starting")
import board
print(dir(board))
from kmk.kmk_keyboard import KMKKeyboard
from kmk.keys import KC
from kmk.scanners import DiodeOrientation
from kmk.modules.layers import Layers
from kmk.extensions.media_keys import MediaKeys

keyboard = KMKKeyboard()
keyboard.modules.append(Layers())
keyboard.extensions.append(MediaKeys())
MOMENTARY = KC.MO(1)
keyboard.col_pins = (board.GP0,board.GP1,board.GP2,board.GP3,board.GP4,board.GP5,board.GP6,board.GP7,board.GP8,board.GP9,board.GP10,board.GP11,board.GP12,)
keyboard.row_pins = (board.GP13,board.GP14,board.GP15,board.GP16,board.GP17,board.GP18,) # unused pins,board.GP19,board.GP20,board.GP21,board.GP22,board.GP26,board.GP27,board.GP28,
keyboard.diode_orientation = DiodeOrientation.COL2ROW

keyboard.keymap = [
        [
KC.ESCAPE,	KC.F1,	KC.F2,	KC.F3,	KC.F4,	KC.F5,	KC.F6,	KC.F7,	KC.F8,	KC.F9,	KC.F10,	KC.F11,	KC.F12,
KC.GRAVE,	KC.N1,	KC.N2,	KC.N3,	KC.N4,	KC.N5,	KC.N6,	KC.N7,	KC.N8,	KC.N9,	KC.N0,	KC.UNDERSCORE,	KC.PLUS,
KC.TAB,	KC.Q,	KC.W,	KC.E,	KC.R,	KC.T,	KC.Y,	KC.U,	KC.I,	KC.O,	KC.P,	KC.LBRACKET,	KC.RBRACKET,
MOMENTARY,	KC.A,	KC.S,	KC.D,	KC.F,	KC.G,	KC.H,	KC.J,	KC.K,	KC.L,	KC.SCOLON,	KC.QUOTE,	KC.BSLASH,
KC.LSHIFT,	KC.Z,	KC.X,	KC.C,	KC.V,	KC.B,	KC.N,	KC.M,	KC.COMMA,	KC.DOT,	KC.SLASH,	KC.RSHIFT,	KC.NO,
KC.NO,	KC.NO,	KC.LGUI,	KC.BSPACE,	KC.SPACE,	KC.LCTRL,	KC.ENTER,	KC.SPACE,	KC.RALT,	KC.NO,	KC.NO,	KC.NO,	KC.NO,
  	  	  	  
        ],
        [
KC.NO,	KC.MEDIA_STOP,	KC.MEDIA_PREV_TRACK,	KC.MEDIA_PLAY_PAUSE,	KC.MEDIA_NEXT_TRACK,	KC.AUDIO_MUTE,	KC.AUDIO_VOL_UP,	KC.KP_SLASH,	KC.KP_SLASH,	KC.KP_ASTERISK,	KC.KP_MINUS,	KC.INSERT,	KC.DELETE,
KC.NO,	KC.NO,	KC.NO,	KC.NO,	KC.NO,	KC.NO,	KC.NO,	KC.KP_7,	KC.KP_8,	KC.KP_9,	KC.KP_PLUS,	KC.HOME,	KC.PGUP,
KC.NO,	KC.NO,	KC.UP,	KC.NO,	KC.NO,	KC.NO,	KC.NO,	KC.KP_4,	KC.KP_5,	KC.KP_6,	KC.END,	KC.UP,	KC.PGDOWN,
KC.NO,	KC.LEFT,	KC.DOWN,	KC.RIGHT,	KC.NO,	KC.NO,	KC.NO,	KC.KP_1,	KC.KP_2,	KC.KP_3,	KC.LEFT,	KC.DOWN,	KC.RIGHT,
KC.NO,	KC.NO,	KC.NO,	KC.NO,	KC.NO,	KC.NO,	KC.NO,	KC.KP_0,	KC.KP_0,	KC.KP_DOT,	KC.KP_ENTER,	KC.NO,	KC.NO,
KC.NO,	KC.NO,	KC.NO,	KC.NO,	KC.NO,	KC.NO,	KC.NO,	KC.NO,	KC.NO,	KC.NO,	KC.NO,	KC.NO,	KC.NO,

]
]
if __name__ == '__main__':
    keyboard.go()

``` 

`

**Comment by sevmeyer** [comment]
Neat! I'm currently working on the PCB for release 1.0, with some changes to the layout. But it should be easy to adapt.
For the Pico, I will include simple firmware written in C, though it will be useful to have configs like this for feature-rich firmware like KMK.

**Reply by wajdzik** [reply]
I'll be happy to do proper Kmk config for it. On daily basis I'm using MS Natural 4000 but i wanted to building something similar for a long time and yours KB is the first 3d printable project that comes with row function keys (I need them for work) that I have found.

One thing - it will be my first own build keyboard as well so my config will not be perfect and there will be bits that I'll not be able to share since I want to have some binds specific for my job.

**Comment by sevmeyer** [comment]
Keep in mind that Chrumm still is an unfinished, mostly untested prototype. I don't think it is a good idea to rely on it for work. Especially if you have no experience building keyboards.

To temper expectations, the next release will abandon the function row in favor of standard Ctrl and arrow keys at the bottom. This will also reduce cost and print time. The function keys are layered instead, closer to the home position:

![chrumm-1 0-layout-preview](https://github.com/sevmeyer/chrumm-keyboard/assets/12496525/546dc65b-bc48-4bdb-9e41-6f543cb37e44)

For hand-wired builds, one can still generate a custom body and firmware, but the PCB will not include a function row.
